homeassistant/components/upnp/device.py

```diff
@@ -140,30 +164,32 @@ def model_name(self):
         except (asyncio.TimeoutError, aiohttp.ClientError, UpnpError):
             _LOGGER.error("Could not delete port mapping")
 
-    async def async_get_total_bytes_received(self):
-        """Get total bytes received."""
-        try:
-            return await self._igd_device.async_get_total_bytes_received()
-        except asyncio.TimeoutError:
-            _LOGGER.warning("Timeout during get_total_bytes_received")
-
-    async def async_get_total_bytes_sent(self):
-        """Get total bytes sent."""
-        try:
-            return await self._igd_device.async_get_total_bytes_sent()
-        except asyncio.TimeoutError:
-            _LOGGER.warning("Timeout during get_total_bytes_sent")
-
-    async def async_get_total_packets_received(self):
-        """Get total packets received."""
-        try:
-            return await self._igd_device.async_get_total_packets_received()
-        except asyncio.TimeoutError:
-            _LOGGER.warning("Timeout during get_total_packets_received")
+    async def async_get_traffic_data(self) -> Mapping[str, any]:
+        """
+        Get all traffic data in one go.
+
+        Traffic data consists of:
+        - total bytes sent
+        - total bytes received
+        - total packets sent
+        - total packats received
+
+        Data is timestamped.
+        """
+        _LOGGER.debug("Getting traffic statistics from device: %s", self)
+
+        values = await asyncio.gather(
+            self._igd_device.async_get_total_bytes_received(),
+            self._igd_device.async_get_total_bytes_sent(),
+            self._igd_device.async_get_total_packets_received(),
+            self._igd_device.async_get_total_packets_sent(),
+            return_exceptions=True,
```

Comment:
Where do we check the exceptions?

Reply:
In the sensor at https://github.com/home-assistant/core/blob/0f7d8acf3cd871bf4ac935434cbe0fb300379452/homeassistant/components/upnp/sensor.py#L143.

But it would be better to let `DataUpdateCoordinator` take care of it and handle the case that `DataUpdateCoordinator.last_update_success` is False. Changed.